### PR TITLE
Changed: reuse puzzle for nft minting, nft_offer by referencing the reuse_public_key_for_change

### PR DESCRIPTION
**Pull request by YeungTing**
### Purpose:
To user, they are not care the actual reason of having new puzzle (increasing deviation index, it just keep make the wallet slower), with reuse_public_key_for_change enabled, still increasing deviation index. so modification completely stop generate new puzzle hash if reuse_public_key_for_change is true for 2 rpc (nft_mint_nft, create_offer_for_ids)

### Current Behavior:
nft_mint_nft, create_offer_for_ids will increase deviation index by 1 for the generate new p2 puzzle hash / p2 inner puzzle.

### New Behavior:
nft_mint_nft, create_offer_for_ids will reuse existing puzzle for the generate new p2 puzzle hash / p2 inner puzzle.

### Testing Notes:
have been tested in mainnet

### CODE REVIEW

**Comment by paninaro** [comment]
@YeungTing Thank you for the fix. I suspect that there may be some tests that need to be updated for this change, but I'll defer to @Quexington for any suggestions in that area.

**Review by ytx1991 — DISMISSED** [review]
looks good, make sure the unit test pass before merging.

**Comment by paninaro** [comment]
Looks like there may be some linting issues. @YeungTing you might want to check the pre-commit section in the contributing guidelines: 
https://github.com/Chia-Network/chia-blockchain/blob/main/CONTRIBUTING.md#pre-commit

Thanks!

**Comment by YeungTing** [comment]
> Looks like there may be some linting issues. @YeungTing you might want to check the pre-commit section in the contributing guidelines: https://github.com/Chia-Network/chia-blockchain/blob/main/CONTRIBUTING.md#pre-commit
> 
> Thanks!

updated for the linting

**Comment by YeungTing** [comment]
> Looks like there may be some linting issues. @YeungTing you might want to check the pre-commit section in the contributing guidelines: https://github.com/Chia-Network/chia-blockchain/blob/main/CONTRIBUTING.md#pre-commit
> 
> Thanks!

Please check again. Thanks

**Comment by YeungTing** [comment]
fixed all linting issue

**Comment by Quexington** [comment]
Hey @YeungTing thanks for fixing the lint!  There's one more requirement we're missing unfortunately, please check this section of the contributing guidlines: https://github.com/Chia-Network/chia-blockchain/blob/main/CONTRIBUTING.md#creating-signed-commits

**Comment by YeungTing** [comment]
> Hey @YeungTing thanks for fixing the lint! There's one more requirement we're missing unfortunately, please check this section of the contributing guidlines: https://github.com/Chia-Network/chia-blockchain/blob/main/CONTRIBUTING.md#creating-signed-commits

I have signed all commits, but seems still not about pass the validator because of the old commit on the history. Could we proceed? Or you prefer need to create new PR from stretch?

**Comment by YeungTing** [comment]
please review

**Comment by Quexington** [comment]
Hey @YeungTing thanks for your contribution!  There was another PR in staging that obsoleted parts of this and it just got merged.  I've recreated your PR over here: https://github.com/Chia-Network/chia-blockchain/pull/16116 just boiled down to the parts that need to change after the merge.  I'll take care of managing CI and getting it merged from this point on thanks!

**Pull request closed without merging**
